synology_dsm/api/core/system.py

```diff
@@ -0,0 +1,118 @@
+# -*- coding: utf-8 -*-
+"""DSM System data and actions."""
+
+
+class SynoCoreSystem(object):
+    """Class containing System data and actions."""
+
+    API_KEY = "SYNO.Core.System"
+
+    def __init__(self, dsm):
+        self._dsm = dsm
+        self._data = {}
+
+    def update(self):
+        """Updates System data."""
+        raw_data = self._dsm.get(self.API_KEY, "info")
+        if raw_data:
+            self._data = raw_data["data"]
+
+    ### get information
+    @property
+    def cpu_clock_speed(self):
+        """Gets System CPU clock speed."""
+        return self._data.get('cpu_clock_speed', {})

```

Comment:
Check all your `.get` defaults, better not adding them than returning the wrong type.

I added some in storage API because I use some properties for `for` loops.

So maybe, remove all of them, and keep only empty lists.

Reply:
ok ... i removed all defaults, except for usb_dev() because return value is a list.